whois/README.md

```diff
@@ -1,13 +1,110 @@
 # polkadot-whois
 
-Tracking of validators using on-chain and substrate telemetry data. The data can be queried by validator's `AccountId` or `(session_index, validator_index)`.
+A tool for converting and querying information about validators using interchangeable unique identifiers, information about a validator can be queried using the following indentifiers:
+- Validator index in `paraSessionInfo` pallet.
+- The account key.
+- The authority discovery id, provided in the hex format, it is the same format as what is displayed in `session.next_keys`.
+- The peer id, the network peer id used for connecting to other nodes.
 
-Usage:
+It prints a list of information for each queried validator, G.g:
+```
+validator_index=<INDEX<>>, account=<ACCOUNT>, peer_id=<PEER_ID>, authorithy_id_discover=0x<AUTHORITY_DISCOVERY>, addresses=<KNOWN_ADDRESSES>, version="POLKADOT_NODE_VERSION"
+```
+
+It uses the subp2p-explorer for querying the DHT information about each validator in the network.
+
+## Usage:
 
+### Query by validator index
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --session-index 9442 \
+ by-validator-index 295 293
+```
+### Query by peer id
+
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --session-index 9442 \
+ by-peer-id 12D3KooWEtD4vrMGsaAmETPx9VXuAu3UyFc7hL92x7ky3TJZwnT7 \
+ 12D3KooWJJC4ACkC6fvsDuAvcgiebPMVNjMsfL3LtgmitAKhC39N
+ 
+```
+
+### Query by authority discovery key
+
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --session-index 9442 \
+ by-authority-discovery 0x1cbbd313b592c053da0dc85fe0ae3c010d7bfc3c858a303418a1707846b6507d \
+ 0xf2fe41ba85a8b16db8642126fd7d4bd3f9cf46c45e9852d528867f3d19474972
+```
+
+### Query by account id
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --session-index 9442 \
+ by-account 5D8DuA8a3obyN6ADUTJPUvx5yj8nnKmE1PqsKBQYCRoiqEak
+```
+### Dump all
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --session-index 9442 \
+ dump-all
+```
+### Usage examples by network.
```

Comment:
What do you think, do we need this section? The previous one could be already very informative.

Reply:
Yes, because it has the bootnodes and rpc nodes in it for each network, so you don't have to search them anywhere else.